blocks/library/categories/index.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+import { Placeholder, Spinner } from 'components';
+import { __ } from 'i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import './block.scss';
+import { registerBlockType } from '../../api';
+import { getCategories } from './data.js';
+import InspectorControls from '../../inspector-controls';
+import ToggleControl from '../../inspector-controls/toggle-control';
+import BlockDescription from '../../block-description';
+import BlockControls from '../../block-controls';
+import BlockAlignmentToolbar from '../../block-alignment-toolbar';
+
+registerBlockType( 'core/categories', {
+	title: __( 'Categories' ),
+
+	icon: 'list-view',
+
+	category: 'widgets',
+
+	defaultAttributes: {
+		showPostCounts: false,
+		displayAsDropdown: false,
+		showHierarchy: false,
+	},
+
+	getEditWrapperProps( attributes ) {
+		const { align } = attributes;
+		if ( 'left' === align || 'right' === align || 'full' === align ) {
+			return { 'data-align': align };
+		}
+	},
+
+	edit: class extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.state = {
+				categories: [],
+			};
+
+			this.categoriesRequest = getCategories();
+
+			this.categoriesRequest
+				.then( categories => this.setState( { categories } ) );
```

Comment:
Sure - it will not be invoked, as the deferred `.then` is executed only if the request is resolved.